### ISSUE TITLE
Compiler doesn't add brackets where necessary

### ISSUE THREAD

**Issue by elliott-v** [Microsoft/TypeScript#12580]
**TypeScript Version:**  2.0.10

**Code**

```ts
var test: boolean = false;
if (test) for (let i = 0; i < 10; i++) {
  let x = () => i;
}

```

**Expected behavior:**
Javascript output that works as intended.

**Actual behavior:**
Javascript output:
```js
var test = false;
if (test)
    var _loop_1 = function(i) {
        var x = function () { return i; };
    };
    for (var i = 0; i < 10; i++) {
        _loop_1(i);
    }
```
The compiler doesn't add brackets for the "if" block, causing this runtime error:
`Uncaught TypeError: _loop_1 is not a function`

**Comment by mhegazy** [comment]
@vladima can you take a look.

**Comment by vladima** [comment]
cannot repro, for both nightly and 2.1 I can see that `then-statement` of `if-statement`  is a block
```
vladima@vladima-HP:~/Sources/play/12580$ cat app.ts 
var test: boolean = false;
if (test) for (let i = 0; i < 10; i++) {
  let x = () => i;
}
vladima@vladima-HP:~/Sources/play/12580$ npm install typescript@next
12580@1.0.0 /home/vladima/Sources/play/12580
└── typescript@2.2.0-dev.20161130 

npm WARN 12580@1.0.0 No description
npm WARN 12580@1.0.0 No repository field.
vladima@vladima-HP:~/Sources/play/12580$ ./node_modules/.bin/tsc --version ; ./node_modules/.bin/tsc app.ts ; cat app.js
Version 2.2.0-dev.20161130
var test = false;
if (test) {
    var _loop_1 = function (i) {
        var x = function () { return i; };
    };
    for (var i = 0; i < 10; i++) {
        _loop_1(i);
    }
}
vladima@vladima-HP:~/Sources/play/12580$ rm app.js
vladima@vladima-HP:~/Sources/play/12580$ npm install typescript@2.1
12580@1.0.0 /home/vladima/Sources/play/12580
└── typescript@2.1.1 

npm WARN 12580@1.0.0 No description
npm WARN 12580@1.0.0 No repository field.
vladima@vladima-HP:~/Sources/play/12580$ ./node_modules/.bin/tsc --version ; ./node_modules/.bin/tsc app.ts ; cat app.js
Version 2.1.1
var test = false;
if (test) {
    var _loop_1 = function (i) {
        var x = function () { return i; };
    };
    for (var i = 0; i < 10; i++) {
        _loop_1(i);
    }
}
vladima@vladima-HP:~/Sources/play/12580$ 
```

**Comment by elliott-v** [comment]
@vladima my mistake.
I didn't specify the right version that I was using: 2.0.10 (the one available on npm).

**Comment by mhegazy** [comment]
the issue seems to be addressed in nightly, and in `typescript@rc`. the fix should be included in the next release of `typescript@2.1.3` or later.

**Comment by richyarD** [comment]
Using typescript@2.6.2 and still got this error.
During build of Angular 2+ app using grunt, grunt-contrib-jshint spots a lot of errors in compiled js code:

`Running "jshint:beforeconcat" (jshint) task
Verifying property jshint.beforeconcat exists in config...OK
Files: src/app/app.component.js, src/app/app.module.js, src/environments/environment.js, src/environments/environment.prod.js, src/main.js, src/polyfills.js -> beforeconcat
Options: force=false, reporter=null, reporterOutputRelative, asi, curly, eqeqeq, eqnull, browser, esversion=6

   src/app/app.component.js
      3 |    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
                                                                                        ^ [W116] Expected '{' andinstead saw 'r'.
      4 |    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c >3 ? d(target, key, r) : d(target, key)) || r;
                  ^ [W116] Expected '{' and instead saw 'for'.
      4 |    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c >3 ? d(target, key, r) : d(target, key)) || r;
                                                                   ^ [W116] Expected '{' and instead saw 'if'.
      4 |    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c >3 ? d(target, key, r) : d(target, key)) || r;
                                                                                        ^ [W084] Expected a conditional expression and instead saw an assignment.
      4 |    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c >3 ? d(target, key, r) : d(target, key)) || r;
                                                                                          ^ [W116] Expected '{' and instead saw 'r'.
   src/app/app.module.js
      3 |    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
                                                                                        ^ [W116] Expected '{' andinstead saw 'r'.
      4 |    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c >3 ? d(target, key, r) : d(target, key)) || r;
                  ^ [W116] Expected '{' and instead saw 'for'.
      4 |    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c >3 ? d(target, key, r) : d(target, key)) || r;
                                                                   ^ [W116] Expected '{' and instead saw 'if'.
      4 |    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c >3 ? d(target, key, r) : d(target, key)) || r;
                                                                                        ^ [W084] Expected a conditional expression and instead saw an assignment.
      4 |    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c >3 ? d(target, key, r) : d(target, key)) || r;
                                                                                          ^ [W116] Expected '{' and instead saw 'r'.

>> 10 errors in 6 files`

I'm aware I could ignore those checks, but I prefer to strictly lint my js code if possible.
